qiskit_optimization/problems/quadratic_program.py

```diff
@@ -1286,89 +939,27 @@ def to_docplex(self) -> Model:
-        objective = self.objective.constant
-        for i, v in self.objective.linear.to_dict().items():
-            objective += v * var[cast(int, i)]
-        for (i, j), v in self.objective.quadratic.to_dict().items():
-            objective += v * var[cast(int, i)] * var[cast(int, j)]
-        if self.objective.sense == QuadraticObjective.Sense.MINIMIZE:
-            mdl.minimize(objective)
-        else:
-            mdl.maximize(objective)
-
-        # add linear constraints
-        for i, l_constraint in enumerate(self.linear_constraints):
-            name = l_constraint.name
-            rhs = l_constraint.rhs
-            if rhs == 0 and l_constraint.linear.coefficients.nnz == 0:
-                continue
-            linear_expr = 0
-            for j, v in l_constraint.linear.to_dict().items():
-                linear_expr += v * var[cast(int, j)]
-            sense = l_constraint.sense
-            if sense == Constraint.Sense.EQ:
-                mdl.add_constraint(linear_expr == rhs, ctname=name)
-            elif sense == Constraint.Sense.GE:
-                mdl.add_constraint(linear_expr >= rhs, ctname=name)
-            elif sense == Constraint.Sense.LE:
-                mdl.add_constraint(linear_expr <= rhs, ctname=name)
-            else:
-                # should never happen
-                raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
-
-        # add quadratic constraints
-        for i, q_constraint in enumerate(self.quadratic_constraints):
-            name = q_constraint.name
-            rhs = q_constraint.rhs
-            if (
-                rhs == 0
-                and q_constraint.linear.coefficients.nnz == 0
-                and q_constraint.quadratic.coefficients.nnz == 0
-            ):
-                continue
-            quadratic_expr = 0
-            for j, v in q_constraint.linear.to_dict().items():
-                quadratic_expr += v * var[cast(int, j)]
-            for (j, k), v in q_constraint.quadratic.to_dict().items():
-                quadratic_expr += v * var[cast(int, j)] * var[cast(int, k)]
-            sense = q_constraint.sense
-            if sense == Constraint.Sense.EQ:
-                mdl.add_constraint(quadratic_expr == rhs, ctname=name)
-            elif sense == Constraint.Sense.GE:
-                mdl.add_constraint(quadratic_expr >= rhs, ctname=name)
-            elif sense == Constraint.Sense.LE:
-                mdl.add_constraint(quadratic_expr <= rhs, ctname=name)
-            else:
-                # should never happen
-                raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
-
-        return mdl
+        warnings.warn(
+            "The to_docplex method is deprecated and will be "
+            "removed in a future release. Instead use the "
+            "export() method",
+            DeprecationWarning,
+        )
+        return self.export()
 
     def export_as_lp_string(self) -> str:
```

Comment:
I am wondering why not keep these methods and put what is in the string support part of export above here and let export just take a translator. The string values there will not autocomplete with an IDE so its a bit more discoverable this way. It would still be done by a translator; but I can understand wanting the symmetry of a  single load and single save like methods.

Reply:
Yes. We should fix it, but I want leave it in this PR. How about discussing it at #77?

Reply:
I tried @stefan-woerner's idea to allow `export('lp')`. https://github.com/Qiskit/qiskit-optimization/pull/122#discussion_r640105391
I see your point. It is hard to discover the method to convert qp to a string with an IDE.
I'm OK to revert this change, i.e., `export` allows only translators as option.

Reply:
I reverted `qp.export_as_lp_string` and `qp.export`. If `qp.export` takes only translators, the impl gets simple. It is also easy for an IDE to discover `export_as_lp_string` too.